packages/valory/skills/abstract_round_abci/tests/test_abci_app_chain.py

```diff
@@ -317,3 +322,67 @@ def test_chain_with_abstract_abci_app_fails(self) -> None:
                 self.round_2c: self.round_3a,
             }
             chain((self.app1_class, self.app2_class, self.app3_class), abci_app_transition_mapping)  # type: ignore
+
+    def test_synchronized_data_type(self) -> None:
+        """Test synchronized data type"""
+
+        abci_app_transition_mapping: AbciAppTransitionMapping = {
+            self.round_1c: self.round_2a,
+            self.round_2c: self.round_1a,
+        }
+
+        sentinel_app1 = object()
+        sentinel_app2 = object()
+
+        def make_sync_data(sentinel: object) -> Type:
+            class SynchronizedData(BaseSynchronizedData):
+                @property
+                def dummy_attr(self) -> object:
+                    return sentinel
+
+            return SynchronizedData
+
+        def make_concrete(round_cls: Type[AbstractRound]) -> Type[AbstractRound]:
+            class ConcreteRound(round_cls):  # type: ignore
+                def check_payload(self, payload: BaseTxPayload) -> None:
+                    pass
+
+                def process_payload(self, payload: BaseTxPayload) -> None:
+                    pass
+
+                def end_block(self) -> None:
+                    pass
+
+                allowed_tx_type = ""
+
+            return ConcreteRound
+
+        sync_data_cls_app1 = make_sync_data(sentinel_app1)
+        sync_data_cls_app2 = make_sync_data(sentinel_app2)
+
+        # don't need to mock all of this since setup creates these classes
+        for abci_app_cls, sync_data_cls in (
+            (self.app1_class, sync_data_cls_app1),
+            (self.app2_class, sync_data_cls_app2),
+        ):
+            for r in abci_app_cls.get_all_rounds():
+                r.synchronized_data_class = sync_data_cls  # type: ignore
+
+        abci_app_cls = chain(
+            (self.app1_class, self.app2_class), abci_app_transition_mapping
+        )
+        synchronized_data = sync_data_cls_app2(db=AbciAppDB(setup_data={}))
+        abci_app = abci_app_cls(synchronized_data, MagicMock(), logging)  # type: ignore
+
+        assert abci_app.initial_round_cls == self.round_1a
+        assert isinstance(abci_app.synchronized_data, sync_data_cls_app1)
+        assert abci_app.synchronized_data.dummy_attr == sentinel_app1  # type: ignore
```

Comment:
setting an instance of `sync_data_cls_app2`, yet converted to an instance of `sync_data_cls_app1`